File: examples/sample-ts-repo/src/greetings.ts
// Pure greeting helpers. No I/O, no logging, no side effects.

export function hello(name: string): string {
    return `Hello, ${name}!`;
}

export function formatWelcome(name: string, locale: string = "en"): string {
    const templates: Record<string, string> = {
        en: "Welcome, {name}.",
        es: "Bienvenido, {name}.",
        fr: "Bienvenue, {name}.",
    };
    const template = templates[locale] ?? templates.en;
    return template.replace("{name}", name);
}

export class Greeter {
    constructor(private salutation: string = "Hello") {}

    greet(name: string): string {
        return `${this.salutation}, ${name}!`;
    }
}

File: examples/sample-py-repo/greetings.py
"""Pure greeting helpers. No I/O, no logging, no side effects."""


def hello(name: str) -> str:
    """Return a simple hello string for ``name``."""
    return f"Hello, {name}!"


def format_welcome(name: str, locale: str = "en") -> str:
    """Return a localized welcome string. Deterministic, pure."""
    templates = {
        "en": "Welcome, {name}.",
        "es": "Bienvenido, {name}.",
        "fr": "Bienvenue, {name}.",
    }
    template = templates.get(locale, templates["en"])
    return template.format(name=name)


class Greeter:
    """Tiny greeter that composes a salutation with a name."""

    def __init__(self, salutation: str = "Hello") -> None:
        self.salutation = salutation

    def greet(self, name: str) -> str:
        """Return ``"<salutation>, <name>!"``."""
        return f"{self.salutation}, {name}!"
